### PR TITLE
match on first location in google search

### LINKED ISSUE #52: Elixir.MatchError api/locations?filter[name]=sidney
https://sentry.io/simplabs/ppm-server/issues/555895296/

```
Elixir.MatchError: no match of right hand side value: %{"results" => [%{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Capital", "short_name" => "Capital", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "British Columbia", "short_name" => "BC", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "Canada", "short_name" => "CA", "types" => ["country", "political"]}], "formatted_address" => "Sidney, BC, Canada", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 48.671867, "lng" => -123.389405}, "southwest" => %{"lat" => 48.631178, "lng" => -123.4179581}}, "location" => %{"lat" => 48.6502411, "lng" => -123.399005}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 48.671867, "lng" => -123.389405}, "southwest" => %{"lat" => 48.631178, "lng" => -123.4179581}}}, "place_id" => "ChIJWSxLC9Fnj1QR8UWcISfWukM", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney Center", "short_name" => "Sidney Center", "types" => ["locality", "political"]}, %{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["administrative_area_level_3", "political"]}, %{"long_name" => "Delaware County", "short_name" => "Delaware County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "New York", "short_name" => "NY", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "13839", "short_name" => "13839", "types" => ["postal_code"]}], "formatted_address" => "Sidney Center, NY 13839, USA", "geometry" => %{"location" => %{"lat" => 42.2906379, "lng" => -75.2557286}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 42.2988914, "lng" => -75.23972119999999}, "southwest" => %{"lat" => 42.2823833, "lng" => -75.27173599999999}}}, "place_id" => "ChIJKyOmm66e24kRq9NFVajuZoI", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Clinton Township", "short_name" => "Clinton Township", "types" => ["administrative_area_level_3", "political"]}, %{"long_name" => "Shelby County", "short_name" => "Shelby County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "Ohio", "short_name" => "OH", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "45365", "short_name" => "45365", "types" => ["postal_code"]}], "formatted_address" => "Sidney, OH 45365, USA", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 40.324103, "lng" => -84.1180299}, "southwest" => %{"lat" => 40.2539969, "lng" => -84.216291}}, "location" => %{"lat" => 40.2842164, "lng" => -84.1554987}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 40.324103, "lng" => -84.1180299}, "southwest" => %{"lat" => 40.2539969, "lng" => -84.216291}}}, "place_id" => "ChIJP_iExpEMP4gRzOa4U9qCKAY", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Richland County", "short_name" => "Richland County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "Montana", "short_name" => "MT", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "59270", "short_name" => "59270", "types" => ["postal_code"]}], "formatted_address" => "Sidney, MT 59270, USA", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 47.7295171, "lng" => -104.1360509}, "southwest" => %{"lat" => 47.6960878, "lng" => -104.206216}}, "location" => %{"lat" => 47.7166836, "lng" => -104.1563253}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{...
  File "lib/airquality/sources/google/geocoding.ex", line 13, in Airquality.Sources.Google.Geocoding.find_location/1
  File "lib/airquality/sources/open_aq.ex", line 7, in Airquality.Sources.OpenAQ.get_locations/1
  File "lib/airquality_web/controllers/location_controller.ex", line 12, in AirqualityWeb.LocationController.index/2
  File "lib/airquality_web/controllers/location_controller.ex", line 1, in AirqualityWeb.LocationController.action/2
  File "lib/airquality_web/controllers/location_controller.ex", line 1, in AirqualityWeb.LocationController.phoenix_controller_pipeline/2
...
(3 additional frame(s) were not displayed)

(MatchError) no match of right hand side value: %{"results" => [%{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Capital", "short_name" => "Capital", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "British Columbia", "short_name" => "BC", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "Canada", "short_name" => "CA", "types" => ["country", "political"]}], "formatted_address" => "Sidney, BC, Canada", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 48.671867, "lng" => -123.389405}, "southwest" => %{"lat" => 48.631178, "lng" => -123.4179581}}, "location" => %{"lat" => 48.6502411, "lng" => -123.399005}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 48.671867, "lng" => -123.389405}, "southwest" => %{"lat" => 48.631178, "lng" => -123.4179581}}}, "place_id" => "ChIJWSxLC9Fnj1QR8UWcISfWukM", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney Center", "short_name" => "Sidney Center", "types" => ["locality", "political"]}, %{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["administrative_area_level_3", "political"]}, %{"long_name" => "Delaware County", "short_name" => "Delaware County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "New York", "short_name" => "NY", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "13839", "short_name" => "13839", "types" => ["postal_code"]}], "formatted_address" => "Sidney Center, NY 13839, USA", "geometry" => %{"location" => %{"lat" => 42.2906379, "lng" => -75.2557286}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 42.2988914, "lng" => -75.23972119999999}, "southwest" => %{"lat" => 42.2823833, "lng" => -75.27173599999999}}}, "place_id" => "ChIJKyOmm66e24kRq9NFVajuZoI", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Clinton Township", "short_name" => "Clinton Township", "types" => ["administrative_area_level_3", "political"]}, %{"long_name" => "Shelby County", "short_name" => "Shelby County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "Ohio", "short_name" => "OH", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "45365", "short_name" => "45365", "types" => ["postal_code"]}], "formatted_address" => "Sidney, OH 45365, USA", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 40.324103, "lng" => -84.1180299}, "southwest" => %{"lat" => 40.2539969, "lng" => -84.216291}}, "location" => %{"lat" => 40.2842164, "lng" => -84.1554987}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 40.324103, "lng" => -84.1180299}, "southwest" => %{"lat" => 40.2539969, "lng" => -84.216291}}}, "place_id" => "ChIJP_iExpEMP4gRzOa4U9qCKAY", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Richland County", "short_name" => "Richland County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "Montana", "short_name" => "MT", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "59270", "short_name" => "59270", "types" => ["postal_code"]}], "formatted_address" => "Sidney, MT 59270, USA", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 47.7295171, "lng" => -104.1360509}, "southwest" => %{"lat" => 47.6960878, "lng" => -104.206216}}, "location" => %{"lat" => 47.7166836, "lng" => -104.1563253}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 47.7295171, "lng" => -104.1360509}, "southwest" => %{"lat" => 47.6960878, "lng" => -104.206216}}}, "place_id" => "ChIJq3eWf4ZxJFMRhAMlC94gOm4", "types" => ["locality", "political"]}, %{"address_components" => [%{"long_name" => "Sidney", "short_name" => "Sidney", "types" => ["locality", "political"]}, %{"long_name" => "Gurley", "short_name" => "Gurley", "types" => ["administrative_area_level_3", "political"]}, %{"long_name" => "Cheyenne County", "short_name" => "Cheyenne County", "types" => ["administrative_area_level_2", "political"]}, %{"long_name" => "Nebraska", "short_name" => "NE", "types" => ["administrative_area_level_1", "political"]}, %{"long_name" => "United States", "short_name" => "US", "types" => ["country", "political"]}, %{"long_name" => "69162", "short_name" => "69162", "types" => ["postal_code"]}], "formatted_address" => "Sidney, NE 69162, USA", "geometry" => %{"bounds" => %{"northeast" => %{"lat" => 41.15650780000001, "lng" => -102.9405759}, "southwest" => %{"lat" => 41.109549, "lng" => -102.998219}}, "location" => %{"lat" => 41.1448219, "lng" => -102.9774497}, "location_type" => "APPROXIMATE", "viewport" => %{"northeast" => %{"lat" => 41.15650780000001, "lng" => -102.9405759}, "southwest" => %{"lat" => 41.109549, "lng" => -102.998219}}}, "place_id" => "ChIJtVzoBn-tcYcR83vYoP_1Pyk", "types" => ["locality", "political"]}], "status" => "OK"}
```

### PR DESCRIPTION
**Pull request by niklaslong**
closes #52

### CODE REVIEW

File: apps/breethe/test/breethe/sources/open_aq_test.exs [hunk 1]
```diff
@@ -154,6 +154,46 @@ defmodule Breethe.Sources.OpenAQTest do
...
+                 "key" => Application.get_env(:breethe, :google_maps_api_key)
+               } == URI.decode_query(conn.query_string)
+
+        response = %{
+          "results" => [
+            %{
+              "geometry" => %{
+                "location" => %{"lat" => 10, "lng" => 20}
+              }
+            }
```
Comment (marcoow) [inline apps/breethe/test/breethe/sources/open_aq_test.exs]: This is only 1 result though 🤔

Comment (niklaslong) [reply apps/breethe/test/breethe/sources/open_aq_test.exs]: 🤦‍♂️

**Comment by niklaslong** [comment]
It's starting to get difficult to test isolated bits of the Geocoding and OpenAQ.{Location, Measurements} modules. We should maybe consider refactoring the tests and splitting them into different files and keep the use of bypass to a minimum (mocks instead where possible)?

**Comment by marcoow** [comment]
Hm, we should think about that maybe but I'd generally like to not over-mock as otherwise we'll run into problems where all of the individual tests with mocked out dependencies work but the individual parts don't work together.

https://i.giphy.com/l0MYSpvx4pnsoMNz2.gif

**Pull request merged**